app/controllers/queue_items_controller.rb

```diff
@@ -21,7 +21,21 @@ def destroy
     item = QueueItem.find(params[:id])
     item.destroy if item && current_user.queue_items.include?(item)
 
-    fix_item_counts
+    current_user.normalize_queue_item_positions
+    redirect_to my_queue_path
+  end
+
+  def update_queue
+    begin
+      fail(StandardError) unless valid_queue_item_data?
+      update_queue_items
+      current_user.normalize_queue_item_positions
+      flash['notice'] = 'Successfully updated My Queue'
+    rescue ActiveRecord::RecordInvalid
+      flash['error'] = 'Queue update failed'
+    rescue StandardError
```

Comment:
By rescuing `StandardError`, you are rescuing pretty much every error. I would recommend creating your own error for this:

```ruby
class InvalidQueueItemData < StandardError
```

You can put that at line 2, then raise that exception instead.